pages/api/users/index.ts

```diff
@@ -0,0 +1,55 @@
+// POST new user to database
+
+// GET user based on id
+
+// GET list of all users
+
+// GET list of all users with a certain role
+
+// PATCH new user information given an id
+
+// GET all staff
+
+// import { object, string, TypeOf } from "yup";
+
+import { NextApiHandler, NextApiRequest, NextApiResponse } from "next";
+import { createUser } from "../../../lib/database/users";
+import { userSchema } from "../../../models/users";
+
+/**
+ * This handles a POST request to /api/users. In Next.js, the file names are what
+ * denotes the route. Read more about requests within Next here:
+ * https://nextjs.org/docs/api-routes/response-helpers
+ *
+ */
```

Comment:
lets make this comment a bit more useful. Just go over that his handles requests to the route